### PR TITLE
feat: document retry-after with 503 (#377)

### LINKED ISSUE #377: Document `Retry-After` with 503
## I suggest

  - a rule to  explicit the usage of `Retry-After` header when 503

## Feedback welcome ;)

### PR DESCRIPTION
**Pull request by tkrop**
Fixes #377.

### CODE REVIEW

**Comment by maxim-tschumak** [comment]
:+1:

File: chapters/http-status-codes-and-errors.adoc [hunk 1]
```diff
@@ -167,8 +167,11 @@ unexpected server execution problem (here, client retry may be sensible)
 |501 |Not Implemented - server cannot fulfill the request (usually
 implies future availability, e.g. new feature). |All
 
-|503 |Service Unavailable - server is (temporarily) not available (e.g.
-due to overload) -- client retry may be sensible. |All
+|503 |Service Unavailable - service is (temporarily) not available (e.g.
+due to overload) -- client retry may be sensible. If possible, the
```
Comment (whiskeysierra) [inline chapters/http-status-codes-and-errors.adoc]: How does this relate to `429 Too Many Requests`?

**Comment by ddufourboivin-zalando** [comment]
👍

File: chapters/http-status-codes-and-errors.adoc (hunk 1 above)
Comment (tkrop) [reply chapters/http-status-codes-and-errors.adoc]: On first sight I would say there is none - at least with respect to issue #377.

Do you suggest to add the same advice for `429`?

Interesting idea to simplify client design. The risk is, that the server might easily create improper advice by always looking up the next expected free slot which may lead to clients requesting to aggressively. Anyhow, lets open a new issue for that.

Comment (whiskeysierra) [reply chapters/http-status-codes-and-errors.adoc]: No, but the example *due to overload* sounds more like 429 to me. I'd expect 503 when one of critical dependencies is down, e.g. the database with a scheduled maintenance window.

Comment (whiskeysierra) [reply chapters/http-status-codes-and-errors.adoc]: Overload due to client requests is a client error, hence 4xx.
Being unable to serve requests due to downstream dependencies is a server error, hence 5xx.

Comment (tkrop) [reply chapters/http-status-codes-and-errors.adoc]: Good point. I will change this.

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by ddufourboivin-zalando** [comment]
👍

**Pull request merged**